### DISCUSSION TITLE
How to upgrade postal 2.1.1 to latest release

### DISCUSSION THREAD

**Discussion by alex-pommier** [postalserver/postal#2488] [Help with using Postal] [answered]
Hello,

I'm not sure what the upgrade steps are to upgrade my current postal installation (2.1.1) to the latest current release. Is is a matter of just replacing the files on the server?

**Comment by willpower232** [comment, marked as answer]
You should be able to run `postal upgrade` from your server and it should do the trick automatically.

https://github.com/postalserver/install/blob/ce3ddfa66e18a7509ebec9c7aebf84fcd70675df/bin/postal#L119

If you have a slightly different configuration to normal then you can just change the version 2.1.1 to whatever the latest is in your hosting configuration but also make sure to run `postal upgrade` inside a container so that the database has any necessary updates.

**Reply by alex-pommier** [reply]
Hi @willpower232 
Thanks for the command that worked perfectly on one of my postal servers, however it fails on a different server and they were both running postal v 2.1.1. The error I got is: 

warning: redirecting to https://github.com/postalserver/install/
From https://postalserver.io/start/install
 * branch            main       -> FETCH_HEAD
There are no candidates for merging among the refs that you just fetched.
Generally this means that you provided a wildcard refspec which had no
matches on the remote end.

**Reply by willpower232** [reply]
if you run `git rev-parse --abbrev-ref HEAD` that should tell you which branch you're on, you should be on main https://github.com/postalserver/install/branches

**Reply by alex-pommier** [reply]
running that command i get: 
fatal: not a git repository (or any of the parent directories): .git

I think i might have a bigger problem, i tried running the commands below and all give me:
INTERNAL ERROR: cannot create temporary directory!

postal status
postal version
etc...

-----------------UPDATE---------------------------
The above is a non-postal issue. The upgrade command works perfectly. I have a separate VM issue that's the reason why the command didn't work for me.

**Reply by willpower232** [reply]
ah sounds dramatic, hope you figured it out
